src/cascadia/TerminalApp/ProfileConversionTraits.cpp

```diff
@@ -0,0 +1,178 @@
+
+    JSON_MAPPINGS(3) = {
+        pair_type{ CloseOnExitAlways, CloseOnExitMode::Always },
+        pair_type{ CloseOnExitGraceful, CloseOnExitMode::Graceful },
+        pair_type{ CloseOnExitNever, CloseOnExitMode::Never },
+    };
+
+    // Override mapping parser to add boolean parsing
+    CloseOnExitMode FromJson(const Json::Value& json)
+    {
+        if (json.isBool())
+        {
+            return json.asBool() ? CloseOnExitMode::Graceful : CloseOnExitMode::Never;
+        }
+        return EnumMapper::FromJson(json);
+    }
+
+    bool CanConvert(const Json::Value& json)
+    {
+        return EnumMapper::CanConvert(json) || json.isBool();
+    }
+};
+
+template<>
+struct JsonUtils::ConversionTrait<winrt::Windows::UI::Text::FontWeight> : public JsonUtils::EnumMapper<unsigned int, JsonUtils::ConversionTrait<winrt::Windows::UI::Text::FontWeight>>
+{
+    // Possible values for Font Weight
+    static constexpr std::string_view FontWeightThin{ "thin" };
+    static constexpr std::string_view FontWeightExtraLight{ "extra-light" };
+    static constexpr std::string_view FontWeightLight{ "light" };
+    static constexpr std::string_view FontWeightSemiLight{ "semi-light" };
+    static constexpr std::string_view FontWeightNormal{ "normal" };
+    static constexpr std::string_view FontWeightMedium{ "medium" };
+    static constexpr std::string_view FontWeightSemiBold{ "semi-bold" };
+    static constexpr std::string_view FontWeightBold{ "bold" };
+    static constexpr std::string_view FontWeightExtraBold{ "extra-bold" };
+    static constexpr std::string_view FontWeightBlack{ "black" };
+    static constexpr std::string_view FontWeightExtraBlack{ "extra-black" };
+
+    // The original parser used the font weight getters Bold(), Normal(), etc.
+    // They were both ugly and *not constant expressions*
+    JSON_MAPPINGS(11) = {
+        pair_type{ FontWeightThin, 100u },
+        pair_type{ FontWeightExtraLight, 200u },
+        pair_type{ FontWeightLight, 300u },
+        pair_type{ FontWeightSemiLight, 350u },
+        pair_type{ FontWeightNormal, 400u },
+        pair_type{ FontWeightMedium, 500u },
+        pair_type{ FontWeightSemiBold, 600u },
+        pair_type{ FontWeightBold, 700u },
+        pair_type{ FontWeightExtraBold, 800u },
+        pair_type{ FontWeightBlack, 900u },
+        pair_type{ FontWeightExtraBlack, 950u },
+    };
+
+    // Override mapping parser to add boolean parsing
+    auto FromJson(const Json::Value& json)
+    {
+        unsigned int value{ 400 };
+        if (json.isUInt())
+        {
+            value = json.asUInt();
+        }
+        else
+        {
+            value = EnumMapper::FromJson(json);
+        }
+
+        winrt::Windows::UI::Text::FontWeight weight{
+            static_cast<uint16_t>(std::clamp(value, 100u, 990u))
+        };
+        return weight;
+    }
+
+    bool CanConvert(const Json::Value& json)
+    {
+        return EnumMapper::CanConvert(json) || json.isUInt();
+    }

```

Comment:
This is maybe my favorite example - this shows how to have a "union" type using the new helpers, and I love it

Reply:
CloseOnExit is an easier union type, but yes, i am so glad this works